### DISCUSSION TITLE
New to contribute to osu-where to start

### DISCUSSION THREAD

**Discussion by LLLgoyour** [ppy/osu#27721] [Q&A] [answered]
Hey guys,

I know it might be silly to ask: I want to contribute to osu (and related projects) for a long while, but I don't know, or I'm hesitating of where to start. Can anyone suggest me something to look on?

About myself: I love playing osu and other rhythm games and I've played a lot. I'm good at solving algorithm problems using Java and cpp, but have little experience to engage in community projects. Plus, I learned c# and don't have experience in building projects.

~~[my osu](https://osu.ppy.sh/u/LLLgoyour)~~

**Comment by Loreos7** [comment]
You can try to fix good first issues https://github.com/ppy/osu/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22

**Comment by Gabixel** [comment, marked as answer]
Hello and welcome! :)
There are several ways you can contribute to the osu! project(s), feel free to choose what suits you:
- Help with the game client/framework:
  - *Note that this repository is for osu!**lazer**, not osu!stable (the current mainstream client, which is feature-locked except in special cases, and it has [its own repo](https://github.com/ppy/osu-stable-issues) for reporting critical-only issues);*
  - You can report bugs or give new ideas for features and design. That's one way to help and contribute. But do it discreetly, and look for existing content. The usual way is to report bugs as *issues*, and give feedback or discuss something using *discussions*;
  - I'd suggest checking out the documentation on the wiki for [`osu` (here)](https://github.com/ppy/osu/wiki) and [`osu-framework`](https://github.com/ppy/osu-framework/wiki), their very cool custom game framework built for lazer (both written in C#). Also check out their READMEs for additional info for debugging locally;
  - You can also join the [peppy's Discord](https://discord.gg/ppy) if you haven't already, where people sometimes ask for help in `general`, or discuss even more about pull requests and issues with the community and the team (also keep up to date by checking the `github-feed`) and get help with code suggestions; <sub>(don't expect much activity over the weekend)</sub>
  - As @Loreos7 suggested, you can also search for `good first issue`s if there are any;
  - There's also the [Crowdin](https://crowdin.com/project/osu-web) project, where you can submit translations for game/website strings (no need to use pull requests);
  - *(Bonus: You can create a third-party gamemode (ruleset) for lazer as a fun experience)*
- Other areas:
  - There's [`osu-web`](https://github.com/ppy/osu-web) for reporting issues with the website or if you also know PHP and want to fix/update stuff;
  - [`osu-wiki`](https://github.com/ppy/osu-wiki) if you want to help translate the [osu! wiki](https://osu.ppy.sh/wiki/en/Main_page) to another language (doesn't use Crowdin);
  - (probably more that I haven't touched, like score processing, resource repo (for game assets), server infrastructure, spectator, etc.)

**Reply by Walavouchey** [reply]
another might be performance point development and discussion ([they have a discord for that](https://discord.gg/aqPCnXu))

> (probably more that I haven't touched, like score processing, resource repo (for game assets), server infrastructure, spectator, etc.)

those repos usually aren't open to contributions, and aren't really the first places to look for places to contribute to. things get added to osu-resources when required by a pr here, almost always after extensive discussion of the new feature/design in question

**Reply by Gabixel** [reply]
Yup, good point (I had left it implicit that those are somewhat complex repos, but it's absolutely right to point that out).
I should also mention the [open projects](https://github.com/ppy/osu/projects?query=is%3Aopen), which are used to track the most important game-side issues the team is working on. So it's fair to say that there are obviously priorities, meaning that pull requests don't always get reviewed immediately (since the team is a small group), but that can also depend on the PR size, of course.

**Reply by bdach** [reply]
where [contributing guidelines](https://github.com/ppy/osu/blob/master/CONTRIBUTING.md)?

the fact that three people have replied here and not one of them linked it (despite it being the actual first thing that should be used for this purpose) makes me just wanna delete it at this point because it's clearly useless

**Reply by Gabixel** [reply]
Weeell... 😳 I admit that I forgot about that; I must have read it a few times and assimilated it quite a bit (but it's actually very useful for this scenario, to avoid creating more wall-of-text).
However, it's also true that I mentioned looking at the READMEs (which include that page), so I hope they're actually read (and that page is still suggested even when creating an issue or pull request), but makes sense to link it in the first place. Completely missed it.

**Reply by LLLgoyour** [reply]
Thanks for your reply! Honestly, I've contributed to the Crowdin project earlier. I'll definitely look at the framework more. Seem like I joined the osu server on discord for a while but never actually engaged in, so I'll follow up on discord for sure.
